Bluewire.Tools.Runner/FindCommits/ResolveCommitFromSemanticVersion.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Threading.Tasks;
+using Bluewire.Common.GitWrapper;
+using Bluewire.Common.GitWrapper.Model;
+using Bluewire.Conventions;
+using Bluewire.Tools.GitRepository;
+using System.Collections.Generic;
+using Bluewire.Tools.Runner.Shared;
+using System;
+
+namespace Bluewire.Tools.Runner.FindCommits
+{
+    public class ResolveCommitFromSemanticVersion : ICommitResolutionJob
+    {
+        private readonly SemanticVersion semVer;
+
+        public ResolveCommitFromSemanticVersion(string semVer)
+        {
+            this.semVer = SemanticVersion.FromString(semVer);
+        }
+        
+        public async Task<Build[]> ResolveCommits(GitSession session, Common.GitWrapper.GitRepository repository)
+        {
+
+            // We expect only BuildNumberOutOfRangeException and BuildNumberNotFoundException exceptions.
+            // The caller of this method could notify the user that no builds were found but doesn't need the specifics.
+
+            if (semVer.IsComplete)
+            {
+                try
+                {
+                    var commit = await FindRemoteCommit(session, repository, semVer);
+                    if (commit != null)
+                        return new Build[1] { new Build() { Commit = commit, SemanticVersion = semVer } };
+                }
+                catch (BuildNumberOutOfRangeException) { }
+                catch (BuildNumberNotFoundException) { }
+                return new Build[0];
+            }
+
+            var refs = new List<Build>(4);
+
+            foreach (string semTag in SemanticVersion.KnownSemanticTags) {

```

Comment:
Brace style?
I should probably standardise our StyleCop setup and add it to `build-scripts` so it can be easily incorporated into all our other repos...